### ISSUE TITLE
Snatcher, minor glitch in the first cutscene of Act 2

### ISSUE THREAD

**Issue by Silanda** [ekeeke/Genesis-Plus-GX#90]
A slightly strange glitch: at the start of the first cutscene of Act 2, when the scene in the Chief's office starts, the Chief's talking head appears too soon, leading to his mouth prematurely making the movements for his first sentence. However, after the premature mouth movements and the initial out of sync audio for them have passed, everything else syncs up properly. The audio for this scene is CD-audio.

FWIW, this is using the libretro core. I can't check this against actual hardware at the moment as I have no way of transferring my save, however, I have no recollection of this ever happening on actual hardware. It doesn't happen using Kega Fusion either.

I've attached my save file. Save slot 1 will start the game right at the scene.

[snatcher_save.zip](https://github.com/ekeeke/Genesis-Plus-GX/files/333349/snatcher_save.zip)

**Comment by Silanda** [comment]
I believe this happens with the NTSC version as well as the PAL.

**Comment by ekeeke** [comment]
I actually noticed the same thing happens in Fusion, using your save, the desynchronization between voice and animation is even worst in Fusion for me. Maybe it's an issue with the CD image I'm using (ISO+WAV with CUE), could you share the cue file you are using, as well as track 8 wav file, somewhere?

**Comment by Silanda** [comment]
No problem, here you go: https://www.dropbox.com/s/dfzznoigt0rcpac/track8%2Bcue.zip?dl=0

I could send you the cue from the bin+cue that I usually use too if you like, as that has track timing info. I'm not having the problem with Fusion even with iso+wav though. Using Genesis Plus GX, I the Chief's talking head appears the instant his head turns in the background graphics. It doesn't do this for me in Fusion, but Fusion does have problems with regards to the track end point.

**Comment by ekeeke** [comment]
The animation is triggered later in Fusion but the audio sync issue is definitively worst with it so I would appreciate a capture from real hardware if you could manage to play up to this point to see when exactly both animation AND audio track are really started (and see which one of the two is happening either too soon or too late in the emulator if they are in sync on real hardware)

**Comment by Silanda** [comment]
I don't have a capture card/device to get video from the real hardware. The best I can do is to record footage with my phone, but I'll make sure that the audio and video sync is fine. I'll try to do that tomorrow.

**Comment by Silanda** [comment]
Awful quality phone footage, but sufficient to show how the game syncs on actual hardware. Made using my original disc to avoid the possibility of a copy causing issues. https://www.youtube.com/watch?v=80b1VILnGRs

**Comment by ekeeke** [comment]
Thanks. It was not an easy one but I eventually found out what was causing the synchronization issue. This is now fixed by https://github.com/ekeeke/Genesis-Plus-GX/commit/38753afaa5e518b985c73410c051965af9ab84bc

For the record, there is some latency before a new track is played caused by CD drive startup time and/or seek time to reach selected CD sector track number. This was already emulated but I found out that track number should also not be updated immediately in CDD status report when a new track is being played, otherwise Snatcher engine will skip some step (and start the bottom-right sprite animation) too fast. Indicating invalid track info (RS1=0xf) for at least the first status report  fixes the synchronization issue in that game.
